### DISCUSSION TITLE
Trailing comma in collection expressions

### DISCUSSION THREAD

**Discussion by ViIvanov** [dotnet/roslyn#71208] [Q&A] [answered]
It looks like the compiler supports trailing commas in collection expressions because the following code compiles successfully ([SharpLab](https://sharplab.io/#v2:EYLgtghglgdgPrALgbQLoAIA2UDOj0C86yAjADToBMFAzBagNxA=)):
```cs
int[] list = [1, 2, 3, ];
// ----------------- ^ trailing comma
```

In the proposal Collection expressions, the syntax [is described as](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#detailed-design):
``` diff
+ collection_expression
  : '[' ']'
  | '[' collection_element ( ',' collection_element )* ']'
  ;

+ collection_element
  : expression_element
  | spread_element
  ;

+ expression_element
  : expression
  ;

+ spread_element
  : '..' expression
  ;
```
I understand it as "After the comma, the `collection_element` is required, that is, in the end, the `expression`."

Does this mean my understanding is incorrect or a different specification was used during development?

**Comment by CyrusNajmabadi** [comment, marked as answer]
A different specification was used during development. We'll fix up the lang spec. Thanks!
